app/controllers/user_posts_controller.rb

```diff
@@ -5,7 +5,66 @@ def index
   end
 
   def show
+    @user, @post = user_post_helper
+  end
+
+  def new
     @user = User.find(params[:user_id])
-    @post = Post.find(params[:post_id])
+    @post = Post.new
+  end
+
+  def create
+    @current_user = User.find(params[:user_id])
+    @post = Post.new(user: @current_user, title: params[:post][:title], text: params[:post][:text])
+    if @post.save
+      flash[:success] = 'Post successfully created'
+      redirect_to "/users/#{@current_user.id}/posts/#{@post.id}"
+    else
+      flash.now[:error] = 'Error: Post could not be created'
+      render :new
+    end
+  end
+
+  def createcomment
+    @user, @post = user_post_helper
+  end
+
+  def submitcomment
+    @user, @post = user_post_helper
+
+    @comment = Comment.create(post_id: @post.id, user_id: @user.id, text: params[:post][:text])
+
+    if @comment.save
+
+      flash[:success] = 'Comment successfully created'
+      redirect_to "/users/#{@user.id}/posts/#{@post.id}"
+    else
+      flash.now[:error] = 'Error: Comment could not be created'
+      render :createcomment
+    end
+  end
+
+  def submitlike
+    @user, @post = user_post_helper
+
+    @like = Like.create(post_id: @post.id, user_id: @user.id)
+
+    if @like.save
+
+      flash[:success] = 'Like successfully created'
+      redirect_to "/users/#{@user.id}/posts/#{@post.id}"
+    else
+      flash.now[:error] = 'Error: Like could not be created'
+      render :show
+    end
+  end
```

Comment:
- [ ] Kindly create a `Likes` controller and then move this action that is related to creating a like to the `Likes` controller okay. 👍  Please remember to create the route and view accordingly.

Reply:
Thanks!! I really appreciate this feedback, the code looks better now!